robot/impl/local_robot.go

```diff
@@ -1329,23 +1329,34 @@ func (r *localRobot) restartSingleModule(ctx context.Context, mod *config.Module
 		}
 	}
 	diff := config.Diff{
-		Left:     r.Config(),
-		Right:    r.Config(),
-		Added:    &config.Config{},
-		Modified: &config.ModifiedConfigDiff{Modules: []config.Module{*mod}},
-		Removed:  &config.Config{},
+		Left:    r.Config(),
+		Right:   r.Config(),
+		Removed: &config.Config{},
+	}
+	if isRunning {
+		diff.Added = &config.Config{}
+		diff.Modified = &config.ModifiedConfigDiff{Modules: []config.Module{*mod}}
+	} else {
+		diff.Added = &config.Config{Modules: []config.Module{*mod}}
```

Comment:
Can this result in two instances of the same module running?

Reply:
did not even think about this case, good catch

it should detect it here and warn
https://github.com/viamrobotics/rdk/blob/main/module/modmanager/manager.go#L261-L264